src/gobstuf/rest/brp/base_view.py

```diff
@@ -108,7 +108,8 @@ def _validate_request_arg(self, arg, value):
         :param value:
         :return:
         """
-        checks = self.request_template.parameter_checks.get(arg)
+        # Use a temporary instance of request_template to get the parameter_checks, hence the (None, None)
+        checks = self.request_template(None, None).parameter_checks.get(arg)
 
```

Comment:
Isn't it easier/better to make parameter_checks a class method?

Reply:
Yes, but we discussed this exact case the other day. The parameter_checks is overridden in the child, that's why this is an instance property.

Thought about creating a get_parameter_checks() class method though, that takes care of initialising with the (None, None) parameters, but I 'm not sure that improves things.

Reply:
What do you think? Should I add a class method get_parameter_checks?

Reply:
You do not need the instance, example:

```
from abc import abstractmethod

class TemplateA:
    parameter_checks = 'Template A'

class TopView:

    @property
    @abstractmethod
    def request_template(self):
        pass

    def validate(self):
        print(self.request_template.parameter_checks)


class ViewA(TopView):
    request_template = TemplateA

tv = ViewA()
tv.validate()

```

Reply:
As far as I can see you do not need to instantiate the request_template. You can simply access any class variable (which parameter_checks is) by the request template property.

Example:
```
from abc import abstractmethod

class TemplateA:

    def __init__(self):
        raise NotImplementedError

    parameter_checks = 'Template A'

class TemplateB:

    def __init__(self):
        raise NotImplementedError

    parameter_checks = 'Template B'

class TopView:

    def __init__(self):
        print("Instantiate TopView")

    @property
    @abstractmethod
    def request_template(self):
        pass

    def validate(self):
        print(self.request_template.parameter_checks)


class ViewA(TopView):

    def __init__(self):
        print("Instantiate top view A")
        super().__init__()

    request_template = TemplateA

class ViewB(TopView):

    def __init__(self):
        print("Instantiate top view B")
        super().__init__()

    request_template = TemplateB

tv = ViewA()
tv.validate()
tv = ViewB()
tv.validate()

```

Reply:
You're right (as discussed on Slack). Already reverted the parameter_checks back to a class property earlier, which I forgot. No need to do weird stuff.